models/ed/R/met2model.ED2.R

```diff
@@ -146,7 +153,7 @@ met2model.ED2 <- function(in.path, in.prefix, outfolder, start_date, end_date, l
     hr   <- NULL
     asec <- sec
     for (y in seq(year, year + nyr - 1)) {
```

Comment:
@mccabete  I'm curious, does this`seq(year, year + nyr - 1)` work for others as it's supposed to? when I try to process years 2005 and 2006, this becomes `2004` and `2005` for the 2005 loop and `2005` and `2006` for the 2006 loop

feel like `nyr <- floor(udunits2::ud.convert(length(sec) * dt, "seconds", "years"))` should be `round` or something?

Reply:
looks like that's due to @ashiklom's [changes](https://github.com/PecanProject/pecan/commit/bd1ce911eb14282caeabe7dfb4affd6714de18d8)

previous version returns a `1`
```
# for an example Ameriflux met `length(sec) = 17520`, `dt = 1800`
floor(length(sec) / 86400 / 365 * dt)
```

new version returns `0`
```
floor(udunits2::ud.convert(length(sec) * dt, "seconds", "years"))
```

Reply:
I don't know why those two would return different values. `sec / 86400 = days` and `days / 365` gives years, so that unit conversion should be the same.

Reply:
Ah, `udunits` is more precise...

```
udunits2::ud.convert(1, "years", "seconds")
# [1] 31556926
86400 * 365
# [1] 31536000
```

So yeah, we should probably change that back. My mistake! I didn't think to check the calculation.

Reply:
A bit more detail: apparently `udunits2` recognizes that an astronomical year is actually ~0.25 days longer than a calendar year.

```
udunits2::ud.convert(1, "years", "days")
# [1] 365.2422
``

Reply:
Indeed, buried in the [UDUnits documentation](https://www.unidata.ucar.edu/software/udunits/udunits-2.0.4/udunits2lib.html) is this:

> CAUTION: The timestamp-unit was created to be analogous to, for example, the degree celsius—but for the time dimension. I've come to believe, however, that creating such a unit was a mistake, primarily because **users try to use the unit in ways for which it was not designed (such as converting dates in a calendar whose year is exactly 365 days long)**. Such activities are much better handled by a dedicated calendar package. Please be careful about using timestamp-units.

Reply:
So, in summary, @mccabete would you mind cleaning up the mess I made by changing `nyr <- floor(...` on line 150 back to `nyr <- floor(length(sec) * dt / 86400 / 365)`?